### ISSUE TITLE
[Feature Request] master events-in port

### ISSUE THREAD

**Issue by ssj71** [falkTX/Carla#351]
I have a midi controller that I want to bind to several fx plugins' controls, but its annoying to have to connect the midi port to each of them. It would be nice if there were an option for a single "carla events-in port" that sends to all plugins that don't have an explicit midi port. It would need to be optional since there are times when you want to connect midi out from one plugin to control another.

**Comment by falkTX** [comment]
this is planned for later, it's how midi-learn will work :)

**Comment by falkTX** [comment]
after thinking about this I don't know if it's a good idea.
because the jack client for the main carla and plugins are different, we have no way to ensure jack is processing the main carla client before the plugins.
ie, jack could very well process the plugins first, and the main carla client later.
or some plugins first, then carla, then the rest of the plugins.
(this will depend on the graph connections)

this leads to inconsistencies. some plugin changes will be made right away while some will have a small delay.

if we also want a global midi-out port, then this becomes impossible.
the same client can't be processed before and after the plugins.
I'd have to create 2 clients, 1 for input and 1 for output...


tbh this sounds like a bad idea when I think more about it.
the way carla works now allows you to send the same exact message to specific plugins without conflicts.
imagine this:
 - plugin 'A' is listening on CC#7, channel 1, for its volume
 - plugin 'B' is doing the same
 - if there was a global midi input port, a CC#7 would control both plugins at once
 - without a global midi in, we can choose where to send that CC#7 into, plus we get the bonus of re-using CC#7 sent by a plugin output port being usable against a different plugin


in short: sorry, but I'll say no to a global midi input port.

BUT, an option to auto-connect ports in some way is possible. that will do what you want without causing any conflicts.
